NPE in loadLicenseInfo when the licenses.xml file contains licenses without <file>

Reported by @TorstenKruse . I think @k-wall mentioned the same thing.

```
[14:34:18] [ERROR] Failed to execute goal org.codehaus.mojo:license-maven-plugin:1.17-SNAPSHOT:aggregate-download-licenses (default-cli) on project pad: Unable to parse license summary output file: e:\Projects_git\pad\target\generated-resources\licenses.xml: NullPointerException -[Help 1]
org.apache.maven.lifecycle.LifecycleExecutionException: Failed to execute goal org.codehaus.mojo:license-maven-plugin:1.17-SNAPSHOT:aggregate-download-licenses (default-cli) on project pad: Unable to parse license summary output file: e:\Projects_git\pad\target\generated-resources\licenses.xml
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:212)
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:153)
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:145)
	at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:116)
	at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:80)
	at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build(SingleThreadedBuilder.java:51)
	at org.apache.maven.lifecycle.internal.LifecycleStarter.execute(LifecycleStarter.java:128)
	at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:307)
	at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:193)
	at org.apache.maven.DefaultMaven.execute(DefaultMaven.java:106)
	at org.apache.maven.cli.MavenCli.execute(MavenCli.java:863)
	at org.apache.maven.cli.MavenCli.doMain(MavenCli.java:288)
	at org.apache.maven.cli.MavenCli.main(MavenCli.java:199)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:564)
	at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced(Launcher.java:289)
	at org.codehaus.plexus.classworlds.launcher.Launcher.launch(Launcher.java:229)
	at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode(Launcher.java:415)
	at org.codehaus.plexus.classworlds.launcher.Launcher.main(Launcher.java:356)
Caused by: org.apache.maven.plugin.MojoExecutionException: Unable to parse license summary output file: e:\Projects_git\pad\target\generated-resources\licenses.xml
	at org.codehaus.mojo.license.AbstractDownloadLicensesMojo.loadLicenseInfo(AbstractDownloadLicensesMojo.java:597)
	at org.codehaus.mojo.license.AbstractDownloadLicensesMojo.execute(AbstractDownloadLicensesMojo.java:334)
	at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo(DefaultBuildPluginManager.java:134)
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:207)
	... 20 more
Caused by: java.lang.NullPointerException
	at java.base/java.io.File.<init>(File.java:359)
	at org.codehaus.mojo.license.AbstractDownloadLicensesMojo.loadLicenseInfo(AbstractDownloadLicensesMojo.java:590)
	... 23 more
```

--- a/src/it/ISSUE-197/invoker.properties
+++ b/src/it/ISSUE-197/invoker.properties
@@ -1,0 +1,1 @@
+invoker.goals = license:download-licenses

--- a/src/it/ISSUE-197/pom.xml
+++ b/src/it/ISSUE-197/pom.xml
@@ -1,0 +1,38 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project xmlns="http://maven.apache.org/POM/4.0.0"
+    xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+    xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/maven-v4_0_0.xsd">
+  <modelVersion>4.0.0</modelVersion>
+
+  <groupId>org.codehaus.mojo.license</groupId>
+  <artifactId>ISSUE-197</artifactId>
+  <version>1.0-SNAPSHOT</version>
+  <name>Integration Test ISSUE-197</name>
+  <url>http://maven.apache.org</url>
+  <description>
+    ISSUE-197
+  </description>
+
+  <dependencies>
+    <dependency>
+      <groupId>commons-logging</groupId>
+      <artifactId>commons-logging</artifactId>
+      <version>1.0</version>
+    </dependency>
+    <dependency>
+      <groupId>org.hamcrest</groupId>
+      <artifactId>hamcrest-api</artifactId>
+      <version>1.0</version>
+    </dependency>
+  </dependencies>
+
+  <build>
+    <plugins>
+      <plugin>
+        <groupId>org.codehaus.mojo</groupId>
+        <artifactId>license-maven-plugin</artifactId>
+        <version>@pom.version@</version>
+      </plugin>
+    </plugins>
+  </build>
+</project>

--- a/src/it/ISSUE-197/postbuild.groovy
+++ b/src/it/ISSUE-197/postbuild.groovy
@@ -1,0 +1,36 @@
+/*
+ * #%L
+ * License Maven Plugin
+ * %%
+ * Copyright (C) 2008 - 2011 CodeLutin, Codehaus, Tony Chemit, Tony chemit
+ * %%
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Lesser Public License for more details.
+ *
+ * You should have received a copy of the GNU General Lesser Public
+ * License along with this program.  If not, see
+ * <http://www.gnu.org/licenses/lgpl-3.0.html>.
+ * #L%
+ */
+
+import java.nio.file.Path;
+import java.nio.file.Files;
+
+Path basePath = basedir.toPath()
+Path licensesXmlPath = basePath.resolve('target/generated-resources/licenses.xml')
+Path aslPath20 = basePath.resolve('target/generated-resources/licenses/apache license 2.0 - license-2.0.txt')
+Path aslPath11 = basePath.resolve('target/generated-resources/licenses/asl-1.1.txt')
+
+assert Files.exists(licensesXmlPath)
+assert Files.exists(aslPath20)
+assert Files.exists(aslPath11)
+
+assert aslPath20.text.contains(' Version 2.0, January 2004')
+assert aslPath11.text.contains('The Apache Software License, Version 1.1')

--- a/src/it/ISSUE-197/prebuild.groovy
+++ b/src/it/ISSUE-197/prebuild.groovy
@@ -1,0 +1,36 @@
+/*
+ * #%L
+ * License Maven Plugin
+ * %%
+ * Copyright (C) 2008 - 2011 CodeLutin, Codehaus, Tony Chemit, Tony chemit
+ * %%
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Lesser Public License for more details.
+ *
+ * You should have received a copy of the GNU General Lesser Public
+ * License along with this program.  If not, see
+ * <http://www.gnu.org/licenses/lgpl-3.0.html>.
+ * #L%
+ */
+
+import java.nio.file.Path;
+import java.nio.file.Files;
+
+Path basePath = basedir.toPath()
+
+basePath.resolve('target').deleteDir()
+
+Path src = basePath.resolve('src/license/licenses.xml')
+Path dest = basePath.resolve('target/generated-resources/licenses.xml')
+Files.createDirectories(dest.getParent())
+Files.copy(src, dest)
+
+println "copied to ${dest}"
+return true

--- a/src/it/ISSUE-197/src/license/licenses.xml
+++ b/src/it/ISSUE-197/src/license/licenses.xml
@@ -1,0 +1,27 @@
+<licenseSummary>
+  <dependencies>
+    <dependency>
+      <groupId>commons-logging</groupId>
+      <artifactId>commons-logging</artifactId>
+      <licenses>
+        <license>
+          <name>Apache License 2.0</name>
+          <url>http://www.apache.org/licenses/LICENSE-2.0.txt</url>
+        </license>
+      </licenses>
+    </dependency>
+    <dependency>
+      <groupId>org.hamcrest</groupId>
+      <artifactId>hamcrest-api</artifactId>
+      <licenses>
+        <license>
+          <!-- In reality, hamcrest-api has BSD license http://www.opensource.org/licenses/bsd-license.php
+               This is just for the sake of testing the overrides -->
+          <name>Apache License 2.0</name>
+          <url>http://www.apache.org/licenses/LICENSE-1.1</url>
+          <file>asl-1.1.txt</file>
+        </license>
+      </licenses>
+    </dependency>
+  </dependencies>
+</licenseSummary>

--- a/src/it/ISSUE-197/src/main/java/org/codehaus/mojo/HelloWorld.java
+++ b/src/it/ISSUE-197/src/main/java/org/codehaus/mojo/HelloWorld.java
@@ -1,0 +1,15 @@
+package org.codehaus.mojo;
+
+public class HelloWorld
+{
+
+    /**
+     * @param args
+     */
+    public static void main( String[] args )
+    {
+        // TODO Auto-generated method stub
+
+    }
+
+}

--- a/src/main/java/org/codehaus/mojo/license/AbstractDownloadLicensesMojo.java
+++ b/src/main/java/org/codehaus/mojo/license/AbstractDownloadLicensesMojo.java
@@ -585,9 +585,16 @@ public abstract class AbstractDownloadLicensesMojo
                 {
                     for ( ProjectLicense license : dep.getLicenses() )
                     {
-                        // Save the URL so we don't download it again
-                        downloadedLicenseURLs.put( license.getUrl(),
-                                new File( licensesOutputDirectory, license.getFile() ) );
+                        final String fileName = license.getFile();
+                        if ( fileName != null )
+                        {
+                            final File licenseFile = new File( licensesOutputDirectory, fileName );
+                            if ( licenseFile.exists() )
+                            {
+                                // Save the URL so we don't download it again
+                                downloadedLicenseURLs.put( license.getUrl(), licenseFile );
+                            }
+                        }
                     }
                 }
             }

